src/urllib3/connection.py

```diff
@@ -482,11 +584,11 @@ def _match_hostname(cert, asserted_hostname):
         )
         # Add cert to exception and reraise so client code can inspect
         # the cert when catching the exception, if they want to
-        e._peer_cert = cert
+        e._peer_cert = cert  # type: ignore
```

Comment:
Should I create a new Exception for this type? Like
```
MatchHostnameError(CertificateError):
    _peer_cert: _PeerCertRetType
```
?

Reply:
I think leaving the `# type: ignore` here for now is fine.